hnn_core/network.py

```diff
@@ -349,10 +349,16 @@ def add_evoked_drive(self, name, *, mu, sigma, numspikes,
             Standard deviation of event time distribution
         numspikes : int
             Number of spikes at each target cell
-        sync_within_trial : bool
-            If True, the target cells receive each numspike synchronously. By
-            default (False), spike times arriving at each target cell are
-            sampled independently using the Gaussian parameteres (mu, sigma).
+        n_drive_cells : int | 'n_cells'
+            The number of drive cells (i.e., ArtificialCell objects) that each
+            contribute an independently sampled synaptic spike to the network
+            according to the Gaussian time distribution (mu, sigma). If
+            'n_cells' (default), a source gets assigned to each of the
+            self.n_cells simulated cells in the network with 1-to-1
+            connectivity. Otherwise, sources are assigned with all-to-all
+            connectivity. If you wish to synchronize the timing of this evoked
+            drive across the network in a given trial with one spike, set
+            n_drive_cells=1.
```

Comment:
here's another thought. How about another parameter `connectivity` or `connectivity_pattern` that defaults to `'one-to-one'` or `'all-to-all'` (or `'full'`?) depending on the drive. That way you make it explicit. And you probably replace `drive['cell_specific']` with `drive['connectivity_pattern']` ?

Reply:
okay as I read the docstring down below, it seems that `cell_specific` is basically doing that. Then, just get rid of the magic and add a `cell_specific` parameter here.

Reply:
I guess I was trying to eliminate unnecessary fine control. We want the user to be able to specify the number of drive cells, but the use-case where `n_drive_cells != 'n_cells'` and `cell_specific=True`  doesn't make sense to me.